### PR TITLE
feat(api): add `ibis.to_sql` to easily get a formatted SQL string

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds a top-level `.sql` API for getting the formatted SQL string from an ibis expression.

### CODE REVIEW

**Comment by kszucs** [comment]
@cpcloud can you add a small test case?

File: ibis/common/pretty.py [hunk 1]
```diff
@@ -49,6 +49,24 @@ def show_sql(
       t0.a * 2 AS c
     FROM t AS t0
     """
+    print(sql(expr, dialect=dialect), file=file)
+
+
+def sql(expr: ir.Expr, dialect: str | None = None) -> str:
```
Comment (jcrist) [inline ibis/common/pretty.py:55]: I'm a bit worried about this being conflated with `Backend.sql` - could we call it `to_sql` instead?

Comment (cpcloud) [reply ibis/common/pretty.py:55]: Sure.

File: docs/api/expressions/top_level.md [hunk 2]
```diff
@@ -28,6 +28,7 @@ These methods and objects are available directly in the `ibis` module.
 ::: ibis.or_
 ::: ibis.param
 ::: ibis.show_sql
+::: ibis.sql
```
Comment (gforsyth) [inline docs/api/expressions/top_level.md:31]:
```suggestion
::: ibis.to_sql
```

**Comment by saulpw** [comment]
I'm not sure about `.sql()` as a name for this, it feels too generic and I can see myself getting confused as to why `.raw_sql()` means "use this sql" while `.sql()` means "return the sql()".

**Comment by cpcloud** [comment]
@saulpw I changed it `to_sql`.

One thing to note is that `.to_sql` is not a method backends. Does that help clear anything up?

**Pull request merged**